src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -1,34 +1,62 @@
 package codeu.controller;
+
 import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
+
 import javax.servlet.RequestDispatcher;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
-public class RegisterServletTest
-{
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+
+public class RegisterServletTest {
   private RegisterServlet registerServlet;
   private HttpServletRequest mockRequest;
   private HttpServletResponse mockResponse;
   private RequestDispatcher mockRequestDispatcher;
- 
+
   @Before
-  public void setup() throws IOException
-  {
+  public void setup() throws IOException {
     registerServlet = new RegisterServlet();
     mockRequest = Mockito.mock(HttpServletRequest.class);
     mockResponse = Mockito.mock(HttpServletResponse.class);
     mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
-    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/register.jsp")).thenReturn(mockRequestDispatcher);
-  }//setup
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/register.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  } //setup
 
   @Test
-  public void testDoGet() throws IOException, ServletException
-  {
+  public void testDoGet() throws IOException, ServletException {
     registerServlet.doGet(mockRequest, mockResponse);
     Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
-  }//testDoGet
-}//RegisterServletTest
+  } //testDoGet
+
+  @Test
+  public void testDoPost_ExistingUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("testing username...");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("testing password...");
+
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.isUserRegistered("testing username...")).thenReturn(true);
+    registerServlet.setUserStore(mockUserStore);
+  } //testDoPost_ExistingUser
+
+  @Test
+  public void testDoPost_NewUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("testing username....");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("testing password....");
+
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.isUserRegistered("testing username....")).thenReturn(false);
+    registerServlet.setUserStore(mockUserStore);
+  } //testDoPost_NewUser
+} //RegisterServletTest
```

Comment:
Ok, same as above comment, you need to test what follows after finding out the user has not been registered. You need to check that a new User has been created, that it has been added to the userStore and also that there's a redirection to the /login page.